### ISSUE TITLE
Blog post contains error for specifying key overrides

### ISSUE THREAD

**Issue by omus** [poulainpi/oryx-with-custom-qmk#11]
The [blog post](https://blog.zsa.io/oryx-custom-qmk-features/) mentions using this snippet for adding a custom `Delete` by using `Shift + Backspace`:
```c
const key_override_t *key_overrides[] = {
    &delete_key_override
};
```

Using this results in the following error:
```
 ./keyboards/zsa/moonlander/keymaps/yrRYo/keymap.c:103:23: error: conflicting types for 'key_overrides'; have 'const key_override_t *[]'
  103 | const key_override_t *key_overrides[] = {
      |                       ^~~~~~~~~~~~~
In file included from quantum/quantum.h:121,
                 from ./.build/obj_zsa_moonlander_yrRYo/src/default_keyboard.h:27,
                 from ./keyboards/zsa/moonlander/keymaps/yrRYo/keymap.c:1:
quantum/process_keycode/process_key_override.h:91:31: note: previous declaration of 'key_overrides' with type 'const key_override_t **'
   91 | extern const key_override_t **key_overrides;
      |                               ^~~~~~~~~~~~~
```

I believe the code which should be used in the blog post is:
```c
const key_override_t **key_overrides = (const key_override_t *[]){
    &delete_key_override
};
```

**Comment by omus** [comment]
The later code, which I took from the screenshot, compiles but doesn't flash correctly on my keyboard. I'll look into this more yet but the inconsistency is still a problem.

**Comment by omus** [comment]
Looks like this worked for me:
```c
const key_override_t **key_overrides = (const key_override_t *[]){
	&delete_key_override,
	NULL // Null terminate the array of overrides!
};
```
– https://github.com/qmk/qmk_firmware/blob/eee0384167b965c60120e1222bc24c0b40cadac4/docs/feature_key_overrides.md#simple-example-idsimple-example
– https://www.reddit.com/r/olkb/comments/1gfvz2a/key_overrides_compilation_error/

Seems like this is at least needed on the moonlander.

**Comment by omus** [comment]
Alright I have a clearer understanding of what's going on. ZSA maintains a [fork of QMK](https://github.com/zsa/qmk_firmware) which at the time of writing the latest version is 0.24. Oryx shows my keyboard firmware version is 0.24.0. The latest version of the QMK firmware is [0.27.1 which uses the newer form for key overrides](https://github.com/qmk/qmk_firmware/blob/0.27.1/docs/features/key_overrides.md#simple-example-simple-example). Since ZSA keyboards normally are limited to [0.24.0 which uses the older form](https://github.com/qmk/qmk_firmware/blob/0.24.0/docs/feature_key_overrides.md).

It appears the key overrides change was made in QMK firmware 0.26.

As this repo is about maintaining Oryx support with QMK features it makes sense to show the older form and the blog post should probably call out the firmware version difference so new users can view the right version of the QMK documentation.

**Comment by JohnLukeBentley** [comment]
@omus Independently I've just observed the same bug and same fix, for the voyager keyboard.

The blog post - [Using a DIY tool to add custom QMK features to your Oryx layout](https://blog.zsa.io/oryx-custom-qmk-features/) - currently contains two code versions of `key_overrides` which you mention ...

In the blog post text ...
```C++
const key_override_t *key_overrides[] = {
	&delete_key_override
};
```

In the blog post image ...
```C++
const key_override_t **key_overrides = (const key_override_t *[]){
    &delete_key_override
};
```

Neither form worked for me. 

I used what might be the same solution (using my actual code that worked with a forward slash rather than delete override) ....

```C++
// Custom QMK here
const key_override_t forwardslash_key_override = ko_make_basic(MOD_MASK_SHIFT, KC_SLASH, KC_LEFT_ANGLE_BRACKET);

// All key overrides to be used
const key_override_t *my_key_overrides[] = {
  &forwardslash_key_override,
  NULL
};

const key_override_t **key_overrides = my_key_overrides;
```

The NULL item, as the last item of the array, seemed significant. 

The solution comes from ChatGPT as it's been a few decades since I used C++. However, I assume the form above, which splits the pointer to the array into separate steps, is equivalent to the single stepped `const key_override_t **key_overrides = (const key_override_t *[]){` form.

In short, all this to confirm the same problem and fix that you detail above.

**Comment by JohnLukeBentley** [comment]
I can confirm the single step form also worked for me ...

```C++
// Custom QMK here
const key_override_t forwardslash_key_override = ko_make_basic(MOD_MASK_SHIFT, KC_SLASH, KC_LEFT_ANGLE_BRACKET);

// All key overrides to be used
const key_override_t **key_overrides = (const key_override_t *[]){
  &forwardslash_key_override,
  NULL // Required as last item in array to prevent compile error
};
```
... so the important difference is indeed the NULL as the last item in the pointer to the array.

**Comment by poulainpi** [comment]
Thanks for your investigations! At the time of writing the blog post, the original solution (the one in the screenshot) worked for me, and on my Voyager, and it continues to work great, so that's a bit strange. However, it seems your solution with the NULL is working for everyone, which is perfect! The ZSA team has already updated the blog with this solution, so I’m closing this issue.

**Comment by JohnLukeBentley** [comment]
Yes strange that the original solution continues to work for you but not for @omus and myself. Agree that with the blog post updated the issue can remain closed. Thanks for checking from your end.
